galaxyproject/galaxy
# Preliminary prebuilt client support

This adds a simple way to install a prebuilt, published galaxy client.  Existing default build and startup process remains unchanged (though this is draft for a sec while I tweak that).

```
make install-client
```

or, with env vars, via run.sh (or common_startup, etc)

```
GALAXY_INSTALL_PREBUILT_CLIENT=1 ./run.sh
```

Note, the latter method will defer to an up-to-date locally built client if it exists (client_build_hash.txt); just a note in case anyone runs into this in testing.



This is not how we want to manage the workspace and multiple packages long term, but we opted for this approach as a way to explore how this *should* work while still solving the problem of quickly plugging in a client for things like `planemo serve`, etc.  This will give us an opportunity to use this during the release cycle and iterate on shipping a prebuilt client before 23.1.  Using npm for distribution of the static build is not optimal -- I want to ultimately tie this into a tagged github release using *that* as a registry that Galaxy can fetch from, but this is an easy first step using infrastructure and tooling we already have.

Updating the client build and publishing will stay a controlled manual process for now, but the gist is that with each point release we'll bump a new version out 23.0.1, 23.0.2, etc,.  This would be automated in the future as a part of a release publication workflow.  The galaxy client version remains unchanged, for right now, but will be set at the freeze, and at that point will follow the galaxy conventions.

Per conversation in UI/UX, this also standardizes watch/serve package script naming to a single 'develop', which is a bit more standard.

## How to test the changes?
(Select all options that apply)
  1. Test above commands, observe galaxy still has a UI.

## License
